CDRIVER-3362 update documentation for non-owning return values (#864)
mongodb/mongo-c-driver: src/libmongoc/doc/mongoc_client_session_get_client.rst

  mongoc_client_t *
  mongoc_client_session_get_client (const mongoc_client_session_t *session);

Returns the :symbol:`mongoc_client_t` from which this session was created. See :symbol:`mongoc_client_start_session()`.

Parameters
----------

* ``session``: A :symbol:`mongoc_client_session_t`.

Returns
-------

> A :symbol:`mongoc_client_t` that is valid only for the lifetime of ``session``.

Author: I investigated further and confirmed your findings. Given `mongoc_client_session_t::client` does not appear to have ownership of the pointed-to `mongoc_client_t` object, would the following be acceptable instead? I would prefer ownership responsibility still be documented:
```suggestion
A :symbol:`mongoc_client_t` that should not be freed.
```

> A :symbol:`mongoc_client_t` that is valid only for the lifetime of ``session``.

Reviewer: Yes, noting that `mongoc_client_t` should not be freed seems like a good idea.
